wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -1,178 +1,909 @@
+First, visit your own forked repo (you can rename the ``ppy`` part in ``https://github.com/ppy/osu-wiki`` with your own GitHub's name for quicker access or click on the "Fork" button in ``ppy/osu-wiki``).
 
+You should be in the ``master`` branch by default.
 
-![GitHub Preview](7.jpg "Github Preview")
+#### Making a new branch
 
-Click on the pencil icon to edit; this should redirect you to the web editor.
+![Slide image about Feature-branch workflow](./img/Contribute_branchworkflow.jpg "Feature-branch workflow")
 
-![GitHub's Web Editor](8.jpg "web Editor Snapshot")
+_[Image taken from Slide 6 from Git branch management by Matt Liu][Matt Liu image source]_
 
-If you want to preview your changes, click the "Preview Changes" tab. To go back and edit, click the "Edit File" tab.
+Frankly speaking, you _could_ do all of your changes in the ``master`` branch. However, it is _bad practice to do this in the long run_.
 
-After making your changes, scroll down and fill the following fields to explain your edit(s) to other contributors.
+This is where branching comes into play.
+By branching, you can keep your original ``master`` branch clean while you make changes in the branches you make.
+In the event of major slip-ups, you can cut off the branch and start a new branch copy based on the clean branch.
 
-![Commit Message](9.jpg "Commit Message")
+![Branch dropdown list](./img/Contribute_webbranch.jpg "Branch dropdown list")
 
-_**Note: Your commit message(s) must thoroughly explain your changes, otherwise, they may not get merged!**_
+Click on the ``Branch: master`` dropdown button.
+There may be some branches shown but the only important branch is the ``master`` branch.
+Type in your new branch name in the textbox shown with translucent text, "Find or create a branch...".
+Hit ``Enter`` key to go to the branch name.
+If the branch name was not found, a new branch based on ``master`` branch will be made.
 
-You would be given an option where to save your edits.
+_Note: Make sure you are in ``Branch: master`` before making a new branch.
+The new branch will always base itself from the previous branch where the branch search was initiated._
+
+Now, see the same dropdown button at the same position again.
+It should be ``Branch: <your branch name>``.
+
+You are now all set to make changes safely with a solid bailout plan.
+
+#### Updating the branch content
+
+To edit a page, go to the folder of that page and select the language of that article.
+From here, you should see something like this:
+
+*Note: Each article is grouped into folders where they contain markdown files arranged by language and their media.
+All article pages inside these folders must use the two letter language codes.
+To know which language code to use, see the [List of ISO 639-1 codes][ISO 639-1] following the `639-1` column.
+Please follow this format when creating a new article.*                
+
+![GitHub Preview](./img/Contribute_webview.jpg "GitHub Preview")
```

Comment:
The image shows the _inside of a file_.

Will be getting a new image instead.

brb.